magento-research/pwa-studio
# [dev]: Refactor Checkout

## Description
Refactoring of checkout components, including updating unit tests. Notably: 

* Removed `SubmitButton` in favor of just using the `Button` component directly.
* Removed the unused `ResetButton`.
* Address Form Refactor
  * fixed bug in checkout reducer where street address was not saved
* BraintreeDropin Refactor
  * The conversion to the mount/unmount logic required some...interesting code. Check out the `useEffect` in the component and see how I had to use a local variable to ensure we teardown/error properly.
* Payment Form Refactor
  * identified a "bug" where `informed` updates state if `initialValues` are used causing multiple renders.
* Form Refactor
  * The parent form is now split into multiple function components. Should we move these into a `Form/` directory and split them out into individual files?


## Related Issue





Closes #1294.

## Verification Steps

1. Go to the checkout form and step through it. Everything should work including all validation, etc.
2. Try doing the checkout form out of order.
3. Try quickly opening and closing sub-forms such as the payments form.
4. Make sure form state is used appropriately. Does it remember address? Does it remember last choices such as "use same as shipping address", etc?

## How have YOU tested this?



Lots of manual testing through different scenarios. See above.

## Screenshots / Screen Captures (if appropriate):

## Proposed Labels for Change Type/Package




## Checklist:
